packages/client/src/runtime/core/compositeProxy/createCompositeProxy.ts

```diff
@@ -39,7 +40,7 @@ const customInspect = Symbol.for('nodejs.util.inspect.custom')
  * @param layers
  * @returns
  */
-export function createCompositeProxy<T extends object>(target: T, layers: CompositeProxyLayer[]): T {
+export function createCompositeProxy<T extends object>(client: Client, target: T, layers: CompositeProxyLayer[]): T {
```

Comment:
I don't think this utility should be aware of `Client` (and the need to unsoundly cast empty objects to `Client` in all tests is a good evidence of this), it's on a different level of abstraction.

Instead, it should take an abstract non-nullable value to get the prototype from:
```suggestion
export function createCompositeProxy<T extends object>(prototypeProvider: {}, target: T, layers: CompositeProxyLayer[]): T {
```

and then on line 177:

```diff
- getPrototypeOf: () => Object.getPrototypeOf(client._originalClient),
+ getPrototypeOf: () => Object.getPrototypeOf(prototypeProvider),
```

Then you can pass `client._originalClient` or `this._originalClient` when you actually do have a client, or an empty object without having to cast anything in the tests.

Reply:
alternatively, you can pass the prototype itself, that should also be fine